pkg/services/cost/aws_ce/controller.go

```diff
@@ -0,0 +1,335 @@
+	"github.com/de-tools/data-atlas/pkg/models/domain"
+	"github.com/de-tools/data-atlas/pkg/services/cost"
+)
+
+var supportedServices = map[string]string{
+	"EC2": "Amazon Elastic Compute Cloud - Compute",
+	"S3":  "Amazon Simple Storage Service",
+	"RDS": "Amazon Relational Database Service",
+}
+
+type controller struct {
+	client *costexplorer.Client
+}
+
+func ControllerFactory(ctx context.Context, profile string) (cost.Controller, error) {
+	cfg, err := LoadConfig(ctx, profile)
+	if err != nil {
+		return nil, fmt.Errorf("unable to load AWS SDK config: %w", err)
+	}
+
+	return &controller{
+		client: costexplorer.NewFromConfig(*cfg),
+	}, nil
+}
+
+func (c *controller) GetSupportedResources() []string {
+	resources := make([]string, 0, len(supportedServices))
+	for service := range supportedServices {
+		resources = append(resources, service)
+	}
+	return resources
+}
+
+func (c *controller) GetRawResourceCost(
+	_ context.Context,
+	resourceType string,
+	days int,
+) ([]domain.ResourceCost, error) {
+	serviceFilter, ok := supportedServices[resourceType]
+	if !ok {
+		return nil, fmt.Errorf("unsupported resource type: %s", resourceType)
+	}
+
+	end := time.Now()
+	start := end.AddDate(0, 0, -days)
+
+	input := &costexplorer.GetCostAndUsageInput{
+		TimePeriod: &types.DateInterval{
+			Start: aws.String(start.Format("2006-01-02")),
+			End:   aws.String(end.Format("2006-01-02")),
+		},
+		Granularity: types.GranularityDaily,
+		Metrics: []string{
+			"UnblendedCost",
+			"UsageQuantity",
+			"NormalizedUsageAmount",
+		},
+		Filter: &types.Expression{
+			And: []types.Expression{
+				{
+					Dimensions: &types.DimensionValues{
+						Key:    types.DimensionService,
+						Values: []string{serviceFilter},
+					},
+				},
+				{
+					Not: &types.Expression{
+						Dimensions: &types.DimensionValues{
+							Key:    types.DimensionRecordType,
+							Values: []string{"Credit", "Refund"},
+						},
+					},
+				},
+			},
+		},
+		GroupBy: []types.GroupDefinition{
+			{
+				Type: types.GroupDefinitionTypeDimension,
+				Key:  aws.String("USAGE_TYPE"),
+			},
+			{
+				Type: types.GroupDefinitionTypeDimension,
+				Key:  aws.String("REGION"),
+			},
+		},
+	}
+
+	result, err := c.client.GetCostAndUsage(context.TODO(), input)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get cost and usage: %w", err)
+	}
+
+	return c.transformCostAndUsageResult(result, resourceType)
+}
+
+func (c *controller) EstimateResourceCost(ctx context.Context, resourceType string, days int) (*domain.Report, error) {
+	costs, err := c.GetRawResourceCost(ctx, resourceType, days)
+	if err != nil {
+		return nil, err
+	}
+
+	report := &domain.Report{
+		Title: fmt.Sprintf("%s Cost Analysis", resourceType),
+		Period: domain.TimePeriod{
+			Start:    time.Now().AddDate(0, 0, -days),
+			End:      time.Now(),
+			Duration: days,
+		},
+		Sections:    make([]domain.ReportSection, 0),
+		TotalAmount: 0,
+		Currency:    "USD",
```

Comment:
We probably need to introduce enums with currency types at some point

Reply:
yeah almost definitely